homeassistant/components/zwave_js/helpers.py

```diff
@@ -344,20 +345,59 @@ def async_get_nodes_from_area_id(
         }
     )
     # Add devices in an area that are Z-Wave JS devices
-    for device in dr.async_entries_for_area(dev_reg, area_id):
-        if next(
-            (
-                config_entry_id
-                for config_entry_id in device.config_entries
-                if cast(
+    nodes.update(
+        async_get_node_from_device_id(hass, device.id, dev_reg)
+        for device in dr.async_entries_for_area(dev_reg, area_id)
+        if any(
+            cast(
+                ConfigEntry,
+                hass.config_entries.async_get_entry(config_entry_id),
+            ).domain
+            == DOMAIN
+            for config_entry_id in device.config_entries
+        )
+    )
+
+    return nodes
+
+
+@callback
+def async_get_nodes_from_label_id(
+    hass: HomeAssistant,
+    label_id: str,
+    ent_reg: er.EntityRegistry | None = None,
+    dev_reg: dr.DeviceRegistry | None = None,
+) -> set[ZwaveNode]:
+    """Get nodes for all Z-Wave JS devices and entities that are in an area."""
+    nodes: set[ZwaveNode] = set()
+    if ent_reg is None:
+        ent_reg = er.async_get(hass)
+    if dev_reg is None:
```

Comment:
This can be written in a single line as: `dev_reg = dev_reg or dr.async_get(hass)`

Reply:
difference of code styles I guess, I find this more readable